type_go18.go

```diff
@@ -248,12 +281,20 @@ func (t *timeType) NewColumnReader(columnIndex, bufferSize int) ColumnReader {
 	}
 }
 
+func (t *timeType) ReadDictionary(columnIndex, numValues int, decoder encoding.Decoder) (Dictionary, error) {
+	if t.Unit.Millis != nil {
```

Comment:
My hunch reading this would be that the `else` block - when millis is nil - is at a lower granularity than this side which then made it confusing why it needs to use more storage. 

Maybe you could use a method call? 

func (t timeType) IsUnix() bool 
func (t timeType) IsPrecise() bool 

or something? I don't know

Reply:
Not super important just was confusing reading it.

Reply:
There's two possible states here: either the millisecond or microsecond units are defined (it's a thrift union underneath), and the parquet format dictates that int32 is used for millisecond precision, int64 otherwise.

I'll add a method like you suggest to improve readability.